### PR TITLE
Introduce the first version of contribution guide

### PR DESCRIPTION
**Pull request by tisonkun**
The first version of the contribution guide consists of:

* Getting Started: set up env, build, and set up IDE.
* Documentation Guide. cc @Anonymitaet 
* Releasing Guide from main repo Wiki.
* Testing and CI. cc @lhotari 
* Brief committer guide.

I'm going to write/migrate more about the developer workflow, community value, internals (threading, protocols, etc.) and so on. Also, a series of legacy documents will be clean up or redirected. But let me show you an early preview and see whether we approve on this initiative and its direction.

cc @merlimat @Anonymitaet @dave2wave @michaeljmarshall @codelipenghui

### CODE REVIEW

**Comment by tisonkun** [comment]
To preview, run `./preview.sh` under `site2/website-next`:


https://user-images.githubusercontent.com/18818196/198513176-2cee07d2-8107-4af0-8d04-d9c1a857c30e.mp4

**Comment by Anonymitaet** [comment]
@tisonkun thanks for moving [various doc contribution guides from pulsar](https://github.com/apache/pulsar/pull/18071) to the Contribution page!

**Comment by urfreespace** [comment]
@tisonkun fantastic, thanks, it's a great job

**Comment by tisonkun** [comment]
ping @merlimat @Anonymitaet @dave2wave @michaeljmarshall @codelipenghui 

I'm going to continue the content and existing materials migration. Waiting for more reviews on this patch.

**Review by Anonymitaet — APPROVED** [review]
LGTM from doc contribution guide pespective

File: site2/website-next/contribute/documentation/contribution.md
Comment (Anonymitaet) [inline site2/website-next/contribute/documentation/contribution.md:161]: @tisonkun FYI: the workflow of this part is changed. I'll update it after this PR gets merged.

**Review by codelipenghui — APPROVED** [review]

File: site2/website-next/contribute/documentation/contribution.md
Comment (tisonkun) [reply site2/website-next/contribute/documentation/contribution.md:161]: Thank you! I notice that @codelipenghui has merged this patch. You can make the change and please inform me when it's ready :)

**Pull request merged**
